Mathlib/Algebra/Group/UniqueProds.lean

```diff
@@ -165,6 +169,29 @@ theorem mulHom_map_iff (f : G ↪ H) (mul : ∀ x y, f (x * y) = f x * f y) :
 #align unique_mul.mul_hom_map_iff UniqueMul.mulHom_map_iff
 #align unique_add.add_hom_map_iff UniqueAdd.addHom_map_iff
 
+open Finset MulOpposite in
+@[to_additive]
+theorem of_mulOpposite
+    (h : UniqueMul (B.map ⟨_, op_injective⟩) (A.map ⟨_, op_injective⟩) (op b0) (op a0)) :
+    UniqueMul A B a0 b0 := by
+  intros a b aA bB ab
+  have := h (mem_map_of_mem _ bB) (mem_map_of_mem _ aA) (by simpa using congr_arg op ab)
+  simpa [and_comm] using this
+
+open Finset MulOpposite in
+@[to_additive]
+theorem iff_mulOpposite :
+    UniqueMul (B.map ⟨_, op_injective⟩) (A.map ⟨_, op_injective⟩) (op b0) (op a0) ↔
+      UniqueMul A B a0 b0 := by
+  refine ⟨of_mulOpposite, fun h ↦ of_mulOpposite (G := MulOpposite G) <| fun a b ha hb hab ↦ ?_⟩
+  simp only [mem_map, Function.Embedding.coeFn_mk, exists_exists_and_eq_and] at ha hb
+  rcases ha with ⟨a, ha, rfl⟩
+  rcases hb with ⟨b, hb, rfl⟩
+  rw [op_inj, op_inj, op_inj, op_inj]
+  apply h ha hb ?_
+  apply_fun op ∘ op using op_injective.comp op_injective
+  exact hab
```

Comment:
For dot notation, I think you might as well extract this reverse direction to a standalone lemma.

Reply:
Add `to_mulOpposite` with the proof and then used `ot/to_mulOpposite` as proofs of the `iff`.